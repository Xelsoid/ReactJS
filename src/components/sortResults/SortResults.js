import PropTypes from "prop-types";
import React from 'react';
import {connect} from "react-redux";
import {bindActionCreators} from "redux";

import Button from '../button/Button';
import './sortResults.scss';
import {sortByRating, sortByDate} from '../../actions/actions';

class SortResults extends React.Component {
  constructor() {
    super();
    this.state = {
      sortingByRating: true,
      sortingByDate: true,
    };
  }

  render() {
    const {movies, sortByDate, sortByRating} = this.props;
    const {sortingByRating, sortingByDate} = this.state;
    if(!movies){return null}

    return(
      <div className="main-sorting">
        <div className="main-sorting__column">
          <span>
            {movies.length}
            {' '}
            movies found
          </span>
        </div>
        <div className="main-sorting__column">
          <span>Sort by:</span>
          <Button
            id='btnreleaseDate'
            title='Release Date'
            disabled={false}
            btnClass='btn--primary'
            callback={()=>{
                this.setState({sortingByDate: !sortingByDate});
                sortByDate(movies, sortingByDate);
              }
            }
          />
          <Button
            id='btnRating'
            title='Rating'
            disabled={false}
            btnClass='btn--primary'
            callback={()=>{
                this.setState({sortingByRating: !sortingByRating});
                sortByRating(movies, sortingByRating);
              }
            }
          />
        </div>
      </div>
  )}
}

function mapStateToProps(state) {
  return {
    movies: state.films
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators({sortByDate, sortByRating}, dispatch)
}

SortResults.defaultProps = {
  movies: {},
  sortByDate: null,
  sortByRating: null,
};

SortResults.propTypes = {
  movies: PropTypes.instanceOf(Object),
  sortByDate: PropTypes.func,
  sortByRating: PropTypes.func,
};

export default connect(mapStateToProps, matchDispatchToProps)(SortResults);
