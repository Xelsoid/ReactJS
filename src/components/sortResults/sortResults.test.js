import React from 'react';
import SortResults from './SortResults';

describe('SortResults', () => {
  it('renders correctly', () => {
    const component = shallow(
      <SortResults />
    );
    expect(component).toMatchSnapshot();
  });
});
