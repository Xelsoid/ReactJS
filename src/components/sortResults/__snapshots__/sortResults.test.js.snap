// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortResults renders correctly 1`] = `
<div
  className="main-sorting"
>
  <div
    className="main-sorting__column"
  >
    <span>
      0
       
      movies found
    </span>
  </div>
  <div
    className="main-sorting__column"
  >
    <span>
      Sort by:
    </span>
    <Button
      btnClass="btn--primary"
      callback={null}
      disabled={false}
      id="btnreleaseDate"
      title="Release Date"
    />
    <Button
      btnClass="btn--primary"
      callback={null}
      disabled={false}
      id="btnRating"
      title="Rating"
    />
  </div>
</div>
`;
